PluginPackage/C#/server.cs

```diff
@@ -0,0 +1,175 @@
+            }
+        }
+    }
+
+    private static void StartServer()
+    {
+        CreateHostBuilder().Build().Run();
+    }
+    private static IHostBuilder CreateHostBuilder(params string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureWebHostDefaults(webBuilder =>
+            {
+                webBuilder.ConfigureKestrel(serverOptions =>
+                {
+                    serverOptions.ConfigureEndpointDefaults(listenOptions =>
+                    {
+                    // Configure endpoint defaults
+                    });
+                })
+                .UseStartup<StartUp>();
+            });
+
+    public static async Task GetToken(string remoteUrl, string responseType, string clientID, string redirectUri, string scope, string localUrl)
+    {
+        Server.redirectUri = redirectUri;
+        Server.clientID = clientID;
+        Server.localUrl = localUrl;
+        Uri uri = new Uri(remoteUrl);
+        string query = "?response_type="+ Uri.EscapeDataString(responseType) + "&client_id=" + Uri.EscapeDataString(clientID) 
+        + "&redirect_uri=" + Uri.EscapeDataString(redirectUri) + "&scope=" + Uri.EscapeDataString(scope);
+        uri = new Uri(uri,query);
+        Task thread = Task.Factory.StartNew(()=>StartServer());
+        OpenBrowser(uri.ToString());
+        await thread;
+    }
+
+    private static void OpenBrowser(string url)
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            url = url.Replace("&", "^&");
+            Process.Start(new ProcessStartInfo("cmd", $"/c start {url}"));
+        }
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            Process.Start("xdg-open", url);
+        }
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            Process.Start("open", url);
+        }
+    }
+
+ /*   public static async Task DownloadFileAsync(string remoteUrl, string localUrl)
+    {
+        using (HttpResponseMessage responseMessage = await client.GetAsync(remoteUrl).ConfigureAwait(false))
+        {
+            if (responseMessage.IsSuccessStatusCode)
+            {
+                var byteArray = await responseMessage.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
+                using (FileStream filestream = new FileStream(localUrl, FileMode.Create, FileAccess.Write, FileShare.None, bufferSize:4096, useAsync:true))
+                {
+                    await filestream.WriteAsync(byteArray, 0, byteArray.Length);
+                }
+            }
+        }        
+    }
+
+    public static async Task DownloadMultipleFilesAsync(string[] remoteUrl, string[] localUrl)
+    {
+        List allTasks = new List();
+        for (int n = 0; n < remoteUrl.Length; n++)
+        {
+            allTasks.Add(DownloadFileAsync(remoteUrl[n], localUrl[n]));
+        }
+        await Task.WhenAll(allTasks).ConfigureAwait(false);
+    }
+    */
+}
+}
```

Comment:
this file needs some cleanup
* commented-out code without explanation
* Ctest namespace doesn't seem useful
* indentation is not consistent